railties/lib/rails/commands/dbconsole/dbconsole_command.rb

```diff
@@ -145,6 +149,9 @@ class DbconsoleCommand < Base # :nodoc:
       class_option :environment, aliases: "-e", type: :string,
         desc: "Specifies the environment to run this console under (test/development/production)."
 
+      class_option :connection, aliases: "-c", type: :string,
+        desc: "Specifies the connection to use.", default: "primary"
+
       def perform
         extract_environment_option_from_argument
```

Comment:
I'd expect to be able to provide the conn as a plain arg. Example scenarios/possibilities:
```bash
rails dbconsole production replica

rails dbconsole production:replica

RAILS_ENV=production rails dbconsole replica

rails dbconsole reporting-replica
```